pacai/ui/topGraphics.py

```diff
@@ -0,0 +1,600 @@
+                                         graphicsConstants.BLACK, graphicsConstants.BLACK)
+
+        coords = (screen_x + ghostSize * (0.3 + dx), screen_y - ghostSize * (0.3 - dy))
+        rightPupil = graphicsUtils.circle((coords), ghostSize * 0.08,
+                                          graphicsConstants.BLACK, graphicsConstants.BLACK)
+
+        ghostImageParts = []
+        ghostImageParts.append(body)
+        ghostImageParts.append(leftEye)
+        ghostImageParts.append(rightEye)
+        ghostImageParts.append(leftPupil)
+        ghostImageParts.append(rightPupil)
+
+        return ghostImageParts
+
+    def moveEyes(self, pos, dir, eyes):
+        (screen_x, screen_y) = (self.to_screen(pos))
+        dx = 0
+        dy = 0
+        if dir == 'North':
+            dy = -0.2
+        if dir == 'South':
+            dy = 0.2
+        if dir == 'East':
+            dx = 0.2
+        if dir == 'West':
+            dx = -0.2
+
+        ghostSize = self.gridSize * graphicsConstants.GHOST_SIZE
+
+        coords = (screen_x + ghostSize * (-0.3 + dx / 1.5), screen_y - ghostSize * (0.3 - dy / 1.5))
+        graphicsUtils.moveCircle(eyes[0], coords, ghostSize * 0.2)
+
+        coords = (screen_x + ghostSize * (0.3 + dx / 1.5), screen_y - ghostSize * (0.3 - dy / 1.5))
+        graphicsUtils.moveCircle(eyes[1], coords, ghostSize * 0.2)
+
+        coords = (screen_x + ghostSize * (-0.3 + dx), screen_y - ghostSize * (0.3 - dy))
+        graphicsUtils.moveCircle(eyes[2], coords, ghostSize * 0.08)
+
+        coords = (screen_x + ghostSize * (0.3 + dx), screen_y - ghostSize * (0.3 - dy))
+        graphicsUtils.moveCircle(eyes[3], coords, ghostSize * 0.08)
+
+    def moveGhost(self, ghost, ghostIndex, prevGhost, ghostImageParts):
+        old_x, old_y = self.to_screen(self.getPosition(prevGhost))
+        new_x, new_y = self.to_screen(self.getPosition(ghost))
+        delta = new_x - old_x, new_y - old_y
+
+        for ghostImagePart in ghostImageParts:
+            graphicsUtils.move_by(ghostImagePart, delta)
+        graphicsUtils.refresh()
+
+        if ghost.scaredTimer > 0:
+            color = graphicsConstants.SCARED_COLOR
+        else:
+            color = graphicsConstants.GHOST_COLORS[ghostIndex]
+        graphicsUtils.edit(ghostImageParts[0], ('fill', color), ('outline', color))
+        self.moveEyes(self.getPosition(ghost), self.getDirection(ghost), ghostImageParts[-4:])
+        graphicsUtils.refresh()
+
+    def getPosition(self, agentState):
+        if (agentState.configuration is None):
+            return (-1000, -1000)
+        return agentState.getPosition()
+
+    def getDirection(self, agentState):
+        if (agentState.configuration is None):
+            return Directions.STOP
+        return agentState.configuration.getDirection()
+
+    def finish(self):
+        # Get the last frame.
+        self.save_frame(force_save = True)
+
+        graphicsUtils.end_graphics()
+
```

Comment:
remove extra spaces

Reply:
Remove all extra spaces or just the ones in the code snippet above? I agree that there are some weird new lines everywhere but I assumed they were for readability/organization purposes.

Reply:
We are trying to make it consistent throughout the codebase, so if there is strange spaces and you can justify removing them, then do.